### PR TITLE
Tests for image utils and color utils

### PR DESCRIPTION
**Pull request by kacperkoza**
# Description

A few tests for ImageUtils and ColorUtils

https://github.com/Swati4star/Images-to-PDF/issues/829

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.
- [X] `./gradlew assembleDebug assembleRelease`
- [X] `./gradlew checkstyle`

# Checklist:
- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings

### CODE REVIEW

File: app/src/test/java/swati4star/createpdf/util/ColorUtilsTest.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        this.mColor1 = mColor1;
+        this.mColor2 = mColor2;
+        this.mIsColorSimilar = mIsColorSimilar;
+    }
+
+    @Test
+    public void colorSimilarCheck() {
+        ColorUtils colorUtils = ColorUtils.getInstance();
+
+        Assert.assertEquals(
```
Comment (Swati4star) [inline app/src/test/java/swati4star/createpdf/util/ColorUtilsTest.java]: add more test cases.

File: app/src/test/java/swati4star/createpdf/util/ImageSortUtilsTest.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        // when
+        invalidSortOption.forEach(sortOption -> {
+                    try {
+                        ImageSortUtils.performSortOperation(sortOption, mImagesPaths);
+                    } catch (Exception ex) {
+                        assertTrue(ex instanceof IllegalArgumentException);
+                        assertTrue(ex.getMessage().startsWith("Invalid sort option"));
+                    }
+                }
+        );
```
Comment (Swati4star) [inline app/src/test/java/swati4star/createpdf/util/ImageSortUtilsTest.java]: add tests for other functions too

File: app/src/main/java/swati4star/createpdf/util/ColorUtils.java [hunk 3]
```diff
@@ -3,14 +3,22 @@
...
-        }
-        return false;
+        double redDiff = diff(Color.red(color1), Color.red(color2));
+        double greenDiff = diff(Color.green(color1), Color.green(color2));
+        double blueDiff = diff(Color.blue(color1), Color.blue(color2));
+        double colorDif = Math.sqrt(redDiff + greenDiff + blueDiff);
+        return colorDif < COLOR_DIFF_THRESHOLD;
+    }
+
+    private double diff(int color, int color2) {
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/util/ColorUtils.java]: add javadoc style comments to this function

**Comment by Swati4star** [comment]
are you working on this @kacperkoza ?

**Comment by kacperkoza** [comment]
@Swati4star yes, just give me one more day :)

File: app/src/main/java/swati4star/createpdf/util/ImageSortUtils.java [hunk 4]
```diff
@@ -72,7 +73,10 @@ private static void sortByDateAsc(ArrayList<String> imagePaths) {
      *
      * @param imagePaths list of image paths to be sorted
      */
-    private static void sortByDateDesc(ArrayList<String> imagePaths) {
+    private static void sortByDateDesc(List<String> imagePaths) {
+        for (String imagePath : imagePaths) {
+            System.out.println(new File(imagePath).lastModified());
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/util/ImageSortUtils.java]: remove this loop

Comment (kacperkoza) [reply app/src/main/java/swati4star/createpdf/util/ImageSortUtils.java]: ofc, that was for debugging purposes

File: app/src/test/java/swati4star/createpdf/util/ColorUtilsTest.java (hunk 1 above)
Comment (Swati4star) [reply app/src/test/java/swati4star/createpdf/util/ColorUtilsTest.java]: are you working on adding this?

Add some negative test cases too

File: app/src/test/java/swati4star/createpdf/util/ImageSortUtilsTest.java [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+
+    private List<String> getFilePaths() {
+        return asList(
+                "src/A-oldest",
+                "src/B-middle",
+                "src/C-latest"
+        );
+    }
+
+    private void createNewFiles() throws Exception {
```
Comment (Swati4star) [inline app/src/test/java/swati4star/createpdf/util/ImageSortUtilsTest.java]: can you also add javadoc style comments to newly added functions?

**Pull request merged**